### PR TITLE
Fix tests in main

### PR DESCRIPTION
**Pull request by aboedo**
Fixes the tests that were failing in `main`, which you can see in https://github.com/RevenueCat/purchases-ios/pull/4172

### CODE REVIEW

**Comment by aboedo** [comment]
@RCGitBot please test

**Comment by aboedo** [comment]
having _any_ references to `ContentUnavailableView` simply won't compile in Xcode 14, since that entity was created using a newer iOS SDK. I'll have to update the code to account for that

**Comment by aboedo** [comment]
@RCGitBot please test

**Comment by aboedo** [comment]
@RevenueCat/coresdk I think for the iOS 14 and 15 tests that are currently failing, we need to update snapshots?

File: .circleci/config.yml [hunk 1]
```diff
@@ -512,7 +512,7 @@ jobs:
   spm-revenuecat-ui-ios-16:
     executor:
       name: macos-executor
-      xcode_version: '14.3.0'
+      xcode_version: '15.4.0'
```
Comment (aboedo) [inline .circleci/config.yml:516]: this one didn't _need_ Xcode 14, and we do intend to move to 15, so I figured it was worth moving it over

File: .circleci/config.yml [hunk 2]
```diff
@@ -653,7 +653,7 @@ jobs:
   run-test-ios-16:
     executor:
       name: macos-executor
-      xcode_version: '14.3.0'
+      xcode_version: '15.4.0'
```
Comment (aboedo) [inline .circleci/config.yml:656]: same here

File: RevenueCat.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -30,6 +30,9 @@
 		2D222BAB27FB7008003D5F37 /* LocalReceiptParserStoreKitTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2D222BAA27FB7008003D5F37 /* LocalReceiptParserStoreKitTests.swift */; };
 		2D22BF6826F3CC6D001AE2F9 /* XCTestCase+Extensions.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2D22BF6626F3CBFB001AE2F9 /* XCTestCase+Extensions.swift */; };
 		2D294E5C26DECFD500B8FE4F /* StoreKit2TransactionListener.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2D294E5B26DECFD500B8FE4F /* StoreKit2TransactionListener.swift */; };
+		2D2AFE8D2C6A834D00D1B0B4 /* TestData.swift in Sources */ = {isa = PBXBuildFile; fileRef = 887A5FDA2C1D037000E1A461 /* TestData.swift */; };
+		2D2AFE8F2C6A9D8700D1B0B4 /* CompatibilityContentUnavailableView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2D2AFE8E2C6A9D8700D1B0B4 /* CompatibilityContentUnavailableView.swift */; };
+		2D2AFE912C6A9EF500D1B0B4 /* Binding+Extensions.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2D2AFE902C6A9EF500D1B0B4 /* Binding+Extensions.swift */; };
```
Comment (aboedo) [inline RevenueCat.xcodeproj/project.pbxproj:35]: some of these files were added into the SPM package but not the Xcode project, so they wouldn't show up in the navigator and they'd fail to compile

File: RevenueCatUI/CustomerCenter/Views/CompatibilityContentUnavailableView.swift [hunk 4]
```diff
@@ -19,7 +19,7 @@ import SwiftUI
 #if os(iOS)
 
 /// A SwiftUI view for displaying a message about unavailable content
-@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(iOS 15.0, *)
 @available(macOS, unavailable)
 @available(tvOS, unavailable)
 @available(watchOS, unavailable)
```
Comment (aboedo) [inline RevenueCatUI/CustomerCenter/Views/CompatibilityContentUnavailableView.swift:25]: this availability was redundant and we were marking it first available and then unavailable for the same platforms

File: RevenueCatUI/CustomerCenter/Views/CompatibilityContentUnavailableView.swift [hunk 5]
```diff
@@ -30,12 +30,20 @@ struct CompatibilityContentUnavailableView: View {
...
     var body: some View {
 
-        if #available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, *) {
-            ContentUnavailableView(
-                title,
-                systemImage: systemImage,
-                description: Text(description)
-            )
+        if #available(iOS 17.0, *) {
+            #if swift(>=5.9)
```
Comment (aboedo) [inline RevenueCatUI/CustomerCenter/Views/CompatibilityContentUnavailableView.swift:34]: this is an old trick that's useful for these Xcode transitions - you point to the min swift version that's packed with the newest xcode, and it effectively works as "if xcode is greater than 15"

File: fastlane/README.md [hunk 6]
```diff
@@ -154,37 +154,13 @@ archive
...
-build CustomEntitlementComputation API tester
-
-### ios build_revenuecatui_api_tester
-
-```sh
-[bundle exec] fastlane ios build_revenuecatui_api_tester
-```
-
-build RevenueCatUI API tester
+run API Tests
```
Comment (aboedo) [inline fastlane/README.md:164]: just automatic and left over from fastlane lane updates from other PRs

File: RevenueCat.xcodeproj/project.pbxproj (hunk 3 above)
Comment (jamesrb1) [reply RevenueCat.xcodeproj/project.pbxproj:35]: Nice find. I wish Xcode had the ability to add all sources in a particular path like SPM allows...

File: RevenueCatUI/CustomerCenter/Views/CompatibilityContentUnavailableView.swift (hunk 5 above)
Comment (jamesrb1) [reply RevenueCatUI/CustomerCenter/Views/CompatibilityContentUnavailableView.swift:34]: Cool!

**Review by jamesrb1 — APPROVED** [review]
LGTM!

File: RevenueCat.xcodeproj/project.pbxproj (hunk 3 above)
Comment (aboedo) [reply RevenueCat.xcodeproj/project.pbxproj:35]: +100

**Review by tonidero — APPROVED** [review]

**Review by vegaro — COMMENTED** [review]
it all makes sense to me

**Comment by aboedo** [comment]
Note: this fixes _most_ tests but it doesn't fix them all, not sure what's happening with iOS 14 and 16 tests. Maybe need to generate snapshots again? 

Merging for the time being since it does leave us in a better place than where we started before the PR

**Pull request merged**